### PR TITLE
Fix string escaping

### PR DESCRIPTION
**Pull request by ParthDesai**
### Description
This PR changes the way we are writing data in csv. Instead of using `"` as quote we are using `$` and use escape character as `\`

### Logic
Since `"` is used in json plenty, we needed a different set of characters to act as quote and escape.

### Justification behind this change
PostgreSql's default quote and escape character both are `"`, which require us to produce ugly json via escaping `"` multiple times, and also creates a problem for string. 

Using `$` as quote enable us to put json as it is, and just escaping `$` character, which results in much cleaner output and also eliminate error with string like `""`.

### Companion PR:
https://github.com/ChorusOne/hipparchus/pull/17

### CODE REVIEW

File: baseapp/baseapp.go [hunk 1]
```diff
@@ -741,29 +741,29 @@ func (app *BaseApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDeliv
...
 		ctx.BlockHeight(),
 		uint32(result.Code),
 		int64(result.GasWanted),
 		int64(result.GasUsed),
-		strings.ReplaceAll(result.Log, "\"", "\"\""),
-		strings.ReplaceAll(string(jsonMemo), "\"", "\"\""),
-		strings.ReplaceAll(string(jsonFee), "\"", "\"\""),
-		strings.ReplaceAll(string(jsonTags), "\"", "\"\""),
-		strings.ReplaceAll(string(jsonMsgs), "\"", "\"\""),
+		strings.ReplaceAll(result.Log, "$", "\\$"),
```
Comment (joe-bowman) [inline baseapp/baseapp.go:762]: This doesn't look right at all; surely all the `strings.ReplaceAll()` calls can be removed now?

**Comment by joe-bowman** [comment]
Apologies for not reviewing this; I missed the email - when you're waiting for a PR can you ping on slack?

**Comment by ParthDesai** [comment]
@joe-bowman reason why we need this is, in case any of strings has `$` sign, it will break.

**Comment by joe-bowman** [comment]
@ParthDesai have you tagged this commit (v0.37.4-16, I think?) and updated dependencies in `hipparchus/go.mod`?

**Comment by ParthDesai** [comment]
@joe-bowman no. Let me do this.

**Pull request merged**